<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>923</width>
    <height>777</height>
   </rect>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>215</red>
        <green>237</green>
        <blue>226</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>215</red>
        <green>237</green>
        <blue>226</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>215</red>
        <green>237</green>
        <blue>226</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>215</red>
        <green>237</green>
        <blue>226</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="font">
   <font>
    <pointsize>20</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Options Calculator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="header">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>10</y>
      <width>291</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>30</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Options Calculator</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonPremiumC">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>390</y>
      <width>321</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Call - Calculate Premium</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonVolatilityC">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>450</y>
      <width>321</width>
      <height>40</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Call - Calculate Volatility</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>70</y>
      <width>321</width>
      <height>261</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>24</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="title">
     <string>Select Action</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QRadioButton" name="rbCP">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>271</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Call - Calculate Premium</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="rbCV">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>271</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Call - Calculate Volatility</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="rbPP">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>251</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Put - Calculate Premium</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="rbPV">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>271</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Put - Calculate Volatility</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButtonPremiumP">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>510</y>
      <width>321</width>
      <height>40</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="text">
     <string>Put - Calculate Premium</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonVolatilityP">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>570</y>
      <width>323</width>
      <height>40</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Put - Calculate Volatility</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonClear">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>650</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>C l e a r</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="premiumLE">
    <property name="geometry">
     <rect>
      <x>704</x>
      <y>553</y>
      <width>176</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="volatilityLabel_2">
    <property name="geometry">
     <rect>
      <x>419</x>
      <y>590</y>
      <width>277</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string> Result - Calculated Volatility:</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QLabel" name="premiumLabel">
    <property name="geometry">
     <rect>
      <x>422</x>
      <y>553</y>
      <width>274</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Result - Calculated Premium:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="volatilityLE_2">
    <property name="geometry">
     <rect>
      <x>704</x>
      <y>590</y>
      <width>176</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="expiratoryDateLabel_2">
    <property name="geometry">
     <rect>
      <x>592</x>
      <y>200</y>
      <width>135</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Expiratory Date</string>
    </property>
   </widget>
   <widget class="QDateEdit" name="expiratoryDateDateEdit">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>200</y>
      <width>174</width>
      <height>29</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
      <horstretch>3</horstretch>
      <verstretch>3</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>27</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>0</width>
      <height>3</height>
     </size>
    </property>
    <property name="baseSize">
     <size>
      <width>0</width>
      <height>27</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
    <property name="inputMethodHints">
     <set>Qt::ImhDate</set>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="correctionMode">
     <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
    </property>
    <property name="dateTime">
     <datetime>
      <hour>0</hour>
      <minute>0</minute>
      <second>0</second>
      <year>2022</year>
      <month>7</month>
      <day>16</day>
     </datetime>
    </property>
    <property name="minimumDate">
     <date>
      <year>2021</year>
      <month>11</month>
      <day>21</day>
     </date>
    </property>
    <property name="currentSection">
     <enum>QDateTimeEdit::DaySection</enum>
    </property>
    <property name="displayFormat">
     <string>dd MMM yyyy</string>
    </property>
    <property name="calendarPopup">
     <bool>true</bool>
    </property>
    <property name="currentSectionIndex">
     <number>0</number>
    </property>
    <property name="date">
     <date>
      <year>2022</year>
      <month>7</month>
      <day>16</day>
     </date>
    </property>
   </widget>
   <widget class="QLabel" name="timeToMaturityLabel">
    <property name="geometry">
     <rect>
      <x>578</x>
      <y>250</y>
      <width>149</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Time to Maturity:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="timeToMaturityLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>250</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="maxLength">
     <number>32767</number>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="premiumInputLabel">
    <property name="geometry">
     <rect>
      <x>572</x>
      <y>300</y>
      <width>155</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Premium Input:  $</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="premiumInputLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>300</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="currentPriceLabel">
    <property name="geometry">
     <rect>
      <x>584</x>
      <y>350</y>
      <width>141</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Current Price:  $</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="currentPriceLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>350</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="strikePriceLabel">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>400</y>
      <width>131</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Strike Price:  $</string>
    </property>
   </widget>
   <widget class="QLabel" name="tickerSymbolLabel">
    <property name="geometry">
     <rect>
      <x>596</x>
      <y>100</y>
      <width>131</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Ticker Symbol:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="tickerLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>100</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="currentDateLabel">
    <property name="geometry">
     <rect>
      <x>609</x>
      <y>150</y>
      <width>118</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Current Date:</string>
    </property>
   </widget>
   <widget class="QDateEdit" name="currentDateDE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>150</y>
      <width>174</width>
      <height>29</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="dateTime">
     <datetime>
      <hour>0</hour>
      <minute>0</minute>
      <second>0</second>
      <year>2022</year>
      <month>7</month>
      <day>16</day>
     </datetime>
    </property>
    <property name="displayFormat">
     <string>dd MMM yyyy</string>
    </property>
    <property name="calendarPopup">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="strikePriceLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>400</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="volatilityLabel_1">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>450</y>
      <width>174</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Volatility Input:  %</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="volatilityLE_1">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>450</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="cursorPosition">
     <number>0</number>
    </property>
    <property name="placeholderText">
     <string>00.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="riskFreeRateLabel">
    <property name="geometry">
     <rect>
      <x>565</x>
      <y>500</y>
      <width>174</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Risk-Free Rate:  %</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="riskFreeRateLE">
    <property name="geometry">
     <rect>
      <x>735</x>
      <y>500</y>
      <width>174</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="placeholderText">
     <string>0.00</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>923</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>rbCP</tabstop>
  <tabstop>rbCV</tabstop>
  <tabstop>rbPP</tabstop>
  <tabstop>rbPV</tabstop>
  <tabstop>tickerLE</tabstop>
  <tabstop>expiratoryDateDateEdit</tabstop>
  <tabstop>timeToMaturityLE</tabstop>
  <tabstop>premiumInputLE</tabstop>
  <tabstop>currentPriceLE</tabstop>
  <tabstop>strikePriceLE</tabstop>
  <tabstop>volatilityLE_1</tabstop>
  <tabstop>riskFreeRateLE</tabstop>
  <tabstop>premiumLE</tabstop>
  <tabstop>volatilityLE_2</tabstop>
  <tabstop>pushButtonPremiumC</tabstop>
  <tabstop>pushButtonVolatilityC</tabstop>
  <tabstop>pushButtonPremiumP</tabstop>
  <tabstop>pushButtonVolatilityP</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
